### PR TITLE
Update secp256k1 submodule

### PR DESCRIPTION
**Pull request by sstone**
Counter-based nonce generation has been extracted into a separate method.

### CODE REVIEW

**Comment by t-bast** [comment]
I don't think we need to merge any update to the libsecp256k1 submodule until they officially ship musig2. I'd rather wait for the musig2 PR to be merged on their side before we update to a "final" version.

**Comment by sstone** [comment]
We can wait. The musig2 branch that we use is behind secp2561k1's musig2 PR but there are only minor changes and an API change (counter-based nonce generation). I hope it will be released soon though :)

**Comment by sstone** [comment]
> Looks mostly good to me, we should make a release now that we depend on an official release of secp256k1! However it looks like you're still pointing the submodule to `jonasnick`'s fork of the repository, can you fix it and point it to https://github.com/bitcoin-core/secp256k1?

Done in https://github.com/ACINQ/secp256k1-kmp/pull/107/commits/6a68a9df89bf640dbb78729ed23b8a9e97de25f3

File: tests/src/commonTest/kotlin/fr/acinq/secp256k1/Musig2Test.kt [hunk 1]
```diff
@@ -90,6 +90,16 @@ class Musig2Test {
         }
     }
 
+    @Test
+    fun `generate secret nonce from counter`() {
```
[the remark this answers is not in the document]
Comment (sstone) [reply tests/src/commonTest/kotlin/fr/acinq/secp256k1/Musig2Test.kt:94]: That was a documentation bug:  the javadoc comment was wrong it was always supposed to be  the signer's public key and that's how we use it in  `bitcoin-kmp` and `lightning-kmp`

**Pull request merged**